compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/MetadataEncoding.hs

```diff
@@ -82,6 +93,40 @@ mapFunDep f (a, b) = (map f a, map f b)
 mapFunDepM :: Monad m => (a -> m b) -> (GHC.FunDep a -> m (GHC.FunDep b))
 mapFunDepM f (a, b) = liftM2 (,) (mapM f a) (mapM f b)
 
+-------------
+-- MINIMAL --
+-------------
+
+minimalName :: LF.TypeSynName -> LF.ExprValName
+minimalName (LF.TypeSynName xs) = LF.ExprValName ("$$minimal" <> T.concat xs)
+
+pattern TStr :: T.Text -> LF.Type
+pattern TStr x = LF.TStruct [(LF.FieldName x, LF.TUnit)]
+
+pattern TCon :: T.Text -> LF.Type -> LF.Type
+pattern TCon a b = LF.TStruct [(LF.FieldName a, b)]
+
+encodeLBooleanFormula :: BF.LBooleanFormula T.Text -> LF.Type
+encodeLBooleanFormula = encodeBooleanFormula . GHC.unLoc
+
+decodeLBooleanFormula :: LF.Type -> Maybe (BF.LBooleanFormula T.Text)
+decodeLBooleanFormula = fmap GHC.noLoc . decodeBooleanFormula
+
+encodeBooleanFormula :: BF.BooleanFormula T.Text -> LF.Type
+encodeBooleanFormula = \case
+    BF.Var x -> TCon "Var" (TStr x)
```

Comment:
I spent an embarrassing amount of time getting confused by the name `TCon` which is something very different from `LF.TCon`. How about calling this something like `EncodedTCon`?

Reply:
Ah, yeah, that is a good point!